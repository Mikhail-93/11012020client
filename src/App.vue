<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>