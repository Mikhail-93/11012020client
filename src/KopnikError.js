class KopnikError extends Error {
    constructor(message, code) {
        super(message)
        this.code = code
    }
}
class KopnikApiError extends KopnikError {
    constructor(message, code, url) {
        super(message, code)
        this.url = url
    }
}



export {KopnikError, KopnikApiError}
