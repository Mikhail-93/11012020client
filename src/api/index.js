import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
