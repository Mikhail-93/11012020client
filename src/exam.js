const cars = {};Debugger attached.
const carHonda = {model: 'CR-V', year: 2015};
const carVolvo = {model: 'XC-60', year: 2018};
const carBMW = {model: 'X5', year: 2019};
cars[carHonda] = 'Good';
cars[carVolvo] = 'Better';
cars[carBMW] = 'The best';
console.log('Honda is', cars[carHonda]);
console.log('Volvo is', cars[carVolvo]);
console.log('BMW is', cars[carBMW]);
