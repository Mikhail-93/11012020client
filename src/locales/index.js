import ru from './ru'
import en from './en'
import sk from './sk.json'

const messages = {
  ru,
  en,
  sk,
}

export default messages
