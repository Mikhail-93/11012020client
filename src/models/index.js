/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}