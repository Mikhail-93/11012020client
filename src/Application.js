import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
