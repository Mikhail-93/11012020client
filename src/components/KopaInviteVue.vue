<template>
    <div class="d-flex flex-wrap justify-start align-center k-kopaInvite"
         :class="{'k-kopaInvite-long': $vuetify.breakpoint.smAndUp}">
        <k-avater v-for="eachAvatar of avatars" :key="eachAvatar.value.id" :value="eachAvatar.value"
                      :size="48" class="ml-1 mr-1 mb-2" :class="eachAvatar.className"
                      :title="eachAvatar.value.rankName"
                      @click="avatar_click(eachAvatar.value)" @dblclick="avatar_dblclick(eachAvatar.value)"
            >
        </k-avater>
        <slot></slot>
    </div>
</template>
<script>
    import _ from 'lodash'
    import {bottle, container} from "../bottle/bottle";
    import i18n from "../plugins/i18n";
    import logger from "./mixin/logger";
    import {Kopa, Kopnik} from "../models";
    import KAvater from "./KAvatar";

    export default {
        name: "KopaInvite",
        mixins: [logger],
        components: {
            KAvater
        },

        data() {
            return {
                application: container.application,
            }
        },
        props: {
            value: {
                type: Kopa
            }
        },
        computed: {
            avatars() {
                const result = this.value.participants
                    .map(eachPart => {
                        return {
                            value: eachPart,
                            className: 'map_avatar' + (this.application.user === eachPart ? ' map_avatar-user' : '') + (this.application.sections.main.selected === eachPart ? ' map_avatar-selected' : ''),
                        }
                    })
                // console.log(result)
                return result
            },
        },
        watch: {},
        methods: {
            avatar_click(event) {
                this.$emit('avatar_click', event)
                return false
            },
            avatar_dblclick(event) {
                this.$emit('avatar_dblclick', event)
                return false
            },
        },
        async created() {

        },
        async mounted() {

        }
    }
</script>
<style lang="scss">
    $avatar-size: 52px; // 48+2*2 borders
    $margin: 4px;
    $element-width: $margin+$avatar-size+$margin;

    .k-kopaInvite {
        width: $element-width*5;
        pointer-events: none;
        /*outline: solid 2px black;*/
    }

    .k-kopaInvite .v-avatar, .k-kopaInvite button {
        pointer-events: all;
    }

    .k-kopaInvite-long {
        width: $element-width*10;
    }

    .k-kopaInvite > .v-avatar {
        transition-property: box-shadow, transform, opacity;
    }
</style>
