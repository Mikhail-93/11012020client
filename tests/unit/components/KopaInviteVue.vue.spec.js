import Vue from 'vue'

import KopaInviteVue from '../../../src/components/KopaInviteVue'
import {Kopnik, Kopa} from "../../../src/models";
import flushPromises from "flush-promises";
import vuePlugins from "../../test-setup";

describe('unit components KopaInviteVue.vue',  () => {
    let vm,
        kopa

    beforeAll(async ()=>{
        await login(1)
    })

    beforeEach(async () => {
        kopa = new Kopa
        vm = new Vue({
            ...KopaInviteVue,
            ...vuePlugins,
            propsData: {
                value: kopa
            }
        })
    })

    it('render', async () => {
        const kopnik2 = Kopnik.getReference(2)

        kopa.add(kopnik2)
        vm.$mount()
        await flushPromises()
        expect(vm.$el).toMatchSnapshot()
    })
})
