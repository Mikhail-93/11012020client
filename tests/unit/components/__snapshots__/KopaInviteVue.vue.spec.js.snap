// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit components KopaInviteVue.vue render 1`] = `
<div
  class="d-flex flex-wrap justify-start align-center k-kopaInvite k-kopaInvite-long"
>
  <span
    class="v-badge ml-1 mr-1 mb-2 v-badge--bottom theme--light map_avatar"
    title="Сысоев Павел Иванович"
  >
    <div
      class="v-avatar"
      style="height: 48px; min-width: 48px; width: 48px;"
    >
      <img
        src="https://lorempixel.com/200/200/?38192"
        style="object-fit: cover;"
      />
    </div>
    <span
      class="v-badge__wrapper"
    >
      <span
        aria-atomic="true"
        aria-label="знак"
        aria-live="polite"
        class="v-badge__badge orange"
        role="status"
      >
        1
      </span>
    </span>
  </span>
   
</div>
`;
